stdlib/public/core/OptionSet.swift

```diff
@@ -286,7 +286,8 @@ extension OptionSet where Element == Self {
   @inlinable // generic-performance
   @discardableResult
   public mutating func remove(_ member: Element) -> Element? {
-    let r = isSuperset(of: member) ? Optional(member) : nil
+    let intersectionElements = intersection(member)
+    let r = !intersectionElements.isEmpty ? Optional(intersectionElements) : nil
     self.subtract(member)
```

Comment:
If `intersectionElements.isEmpty` then surely we can skip the subtraction, no?

Reply:
Yes.
But I prefer to keep code clean and don't add this condition statement at here, and follow other methods of OptionSet

Reply:
I agree with Lily -- an early exit with a `guard` would read best here.

Reply:
@lorentey 
Thanks for your review. I have changed code at [d5e2533](https://github.com/apple/swift/pull/28378/commits/d5e2533ceff38815c15f89a44048393d501ce439)